ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/KsqlUncaughtExceptionHandler.java

```diff
@@ -15,23 +15,41 @@
 
 package io.confluent.ksql.rest.util;
 
+import com.google.common.annotations.VisibleForTesting;
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 import io.confluent.ksql.rest.server.KsqlServerMain;
+import org.apache.kafka.streams.processor.internals.StreamThread;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 
 public class KsqlUncaughtExceptionHandler implements Thread.UncaughtExceptionHandler {
   private final Runnable flusher;
+  private final String streamsThreadMessage;
 
   private static final Logger log = LoggerFactory.getLogger(KsqlServerMain.class);
 
   public KsqlUncaughtExceptionHandler(final Runnable flusher) {
+    this(flusher, "");
+  }
+
+  @VisibleForTesting
+  public KsqlUncaughtExceptionHandler(
+      final Runnable flusher,
+      final String streamsThreadMessage
+  ) {
     this.flusher = flusher;
+    this.streamsThreadMessage = streamsThreadMessage;
```

Comment:
Thanks for addressing the comments! Is there any reason not to use a `CountDownLatch` here instead of a message that we want to print out? I just think that'll be a little cleaner (avoids the assert eventually call and reading to stdout)

Reply:
@stevenpyzhang Wrote this test, not trying to take credit for his work. I just did the 3 line fix. But my local build it messed up. It might be easier for Steven to just push his changes and open a new pr at this point...

Reply:
it's latched now